bin/hdfcoinc/pycbc_page_ifar

```diff
@@ -157,15 +157,27 @@ expected_cumnum = fp.attrs['foreground_time'] / expected_ifar / lal.YRJUL_SI
 # get background timeslide IDs and IFAR values
 
 # Use these backgrounds for n-th stage of hierarchical removals
-# and if the open box option
-if h_inc_back_num > 0 and opts.open_box:
-    back_tsid = fp['background_h%s/timeslide_id' % h_inc_back_num][:]
-    back_ifar = fp['background_h%s/ifar' % h_inc_back_num][:]
+# and test for backwards compatibility and that stage was written. 
+if opts.open_box:
+    if h_inc_back_num >= 0 and h_iterations is not None and h_iterations != 0:
+        back_tsid = fp['background_h%s/timeslide_id' % h_inc_back_num][:]
+        back_ifar = fp['background_h%s/ifar' % h_inc_back_num][:]
+
+    # If the user wants to plot foreground events at h_inc_back_num == 0
+    # and h_iterations == 0 then no hierarchical removals were done and
+    # the line below is safe to use. The other case is the trigger-file
+    # is before hierarchical removals so the else below catches that.
+    # The third case is requesting to plot hierarchical removal level
+    # greater than was done and an empty plot is made earlier in the code.
+    else :
+        back_tsid = fp['background/timeslide_id'][:]
+        back_ifar = fp['background/ifar'][:]
+
 else :
     # If the box is closed use the background from no hierarchical
     # removals. Results should be agnostic of judgments about
     # whether a hierarchical removal was done.
-    if not opts.open_box and h_iterations > 0:
+    if h_iterations > 0 and h_iterations is not None:
         back_tsid = fp['background_h0/timeslide_id'][:]
         back_ifar = fp['background_h0/ifar'][:]
 
```

Comment:
The lines after this are:
```
    else :
        back_tsid = fp['background/timeslide_id'][:]
        back_ifar = fp['background/ifar'][:]
```
This catches the case for closed box and the statmap file is either not backwards compatible or no hierarchical removals were done.